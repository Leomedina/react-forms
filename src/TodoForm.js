import React, { useState } from 'react';

const TodoForm = ({ addTodo }) => {
  const [formData, setFormData] = useState({ task: "" });

  const handleChange = e => {
    const { name, value } = e.target;
    setFormData(data => ({
      ...data,
      [name]: value
    }));
  };

  const handleSubmit = e => {
    e.preventDefault();
    addTodo({ ...formData });
    setFormData({ task: "" })
  }

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="task">Todo:</label>
      <input
        id="task"
        type="text"
        name="task"
        value={formData.task}
        onChange={handleChange}
      />
      <button>Add</button>
    </form >
  );
}

export default TodoForm;