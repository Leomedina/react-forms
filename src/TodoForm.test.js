import React from "react";
import { render } from "@testing-library/react";
import TodoForm from './TodoForm';

describe("Smoke and snapshot tests", function () {
  it("should render without crashing", function () {
    const addTodo = () => null;
    render(<TodoForm addTodo={addTodo} />);
  });

  it("should match snapshot", function () {
    const addTodo = () => null;
    const { asFragment } = render(<TodoForm addTodo={addTodo} />);
    expect(asFragment()).toMatchSnapshot();
  });
});