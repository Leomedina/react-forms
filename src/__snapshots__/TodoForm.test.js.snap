// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      Todo:
    </label>
    <input
      id="task"
      name="task"
      type="text"
      value=""
    />
    <button>
      Add
    </button>
  </form>
</DocumentFragment>
`;
